contributing/building-a-bento-amp-extension.md

```diff
@@ -0,0 +1,681 @@
+    cleanupIssue: 'https://github.com/ampproject/amphtml/issues/XXXYYY',
+  },
+  // ...
+];
+```
+
+Then protect your code with a check for the component-specific flag `isExperimentOn(win, 'bento-my-element')`, or the global flag which enables all Bento components `isExperimentOn(win, 'bento')`, and only execute your code when it is on.
+
+```javascript
+import {CSS} from '../../../build/amp-my-element-0.1.css';
+import {isExperimentOn} from '../../../src/experiments';
+import {userAssert} from '../../../src/log';
+
+/** @const */
+const TAG = 'amp-my-element';
+
+Class AmpMyElement extends AMP.PreactBaseElement {
+  /** @override */
+  isLayoutSupported(layout) {
+    userAssert(
+      isExperimentOn(this.win, 'bento') ||
+        isExperimentOn(this.win, 'bento-my-element'),
+      'expected global "bento" or specific "bento-my-element" experiment to be enabled'
+    );
+    return layout == LAYOUT.FIXED;
+  }
+}
+
+AMP.extension('amp-my-element', '0.1', AMP => {
+  AMP.registerElement('amp-my-element', AmpMyElement, CSS);
+});
+```
+
+### Enabling and removing your experiment
+
+Users wanting to experiment with your element can then go to the
+[experiments page](https://cdn.ampproject.org/experiments.html) and
+enable your experiment.
+
+If you are testing on your localhost, use the command `AMP.toggleExperiment(id, true/false)` to enable the experiment. One option is to use `AMP.toggleExperiment('bento', true)` to enable all Bento components at once.
+
+File a GitHub issue to cleanup your experiment. Assign it to yourself as a reminder to remove your experiment and code checks. Removal of your experiment happens after the extension has been thoroughly tested and all issues have been addressed.
+
+## Documenting your element
+
+Create a .md file that serves as the main documentation for your element. This document should include:
+
+-   Summary table
+-   Overview
+-   How to use it including code snippets and images
+-   Examples
+-   Attributes to specify (optional and required)
+-   Validation
+
+For samples of element documentation, see: [amp-accordion](https://github.com/ampproject/amphtml/blob/master/extensions/amp-accordion/amp-accordion.md), [amp-instagram](https://github.com/ampproject/amphtml/blob/master/extensions/amp-instagram/amp-instagram.md), [amp-stream-gallery](https://github.com/ampproject/amphtml/blob/master/extensions/amp-stream-gallery/amp-stream-gallery.md)
+
+Note that for Bento upgrades of existing AMP extensions, a `Migration notes` section is required to detail any differences between the newer Bento and prior versions.
+
+## Example of using your extension
+
+This greatly helps users to understand and demonstrate how
+your element works, and provides an easy start-point for them to
+experiment with it. This is basically where you actually build AMP
+HTML and non-AMP HTML documents and use your element in it by creating a file in the local `storybook/` directory, usually with the `Basic.amp.js` file
```

Comment:
Let's add a subsection on how to run the storybooks.

Reply:
Done.